### PR TITLE
upgrade to yesod-test-1.5

### PR DESCRIPTION
**Pull request by gregwebs**
Integration tests will perform application logging to stdout
This means you can actually figure out why your tests are failing.

A downside is a lot of output interleaved with your tests.
We should address that downside in a future change.
I already have something working locally that I want to try out
for a few weeks and refine a bit.

### CODE REVIEW

**Comment by MaxGabriel** [comment]
Getting a few compiler errors running this; investigating if that's just weirdness in my local environment or not. Here's the output: https://gist.github.com/MaxGabriel/766f5490d9985b07dd91

[stackage-nightly](http://www.stackage.org/nightly-2015-08-21) doesn't have yesod-test-1.5 but I imagine that it will by the time the version of yesod-bin this goes out with is in stackage.

File: test/TestImport.hs [hunk 1]
```diff
@@ -31,7 +31,8 @@ withApp = before $ do
         ignoreEnv
     foundation <- makeFoundation settings
     wipeDB foundation
-    return foundation
+    logWare <- liftIO $ makeLogWare foundation
```
Comment (MaxGabriel) [inline test/TestImport.hs]: `makeLogWare` needs to be imported

File: Application.hs [hunk 2]
```diff
@@ -79,7 +80,14 @@ makeFoundation appSettings = do
 -- applying some additional middlewares.
 makeApplication :: App -> IO Application
 makeApplication foundation = do
-    logWare <- mkRequestLogger def
+    logWare <- makeLogWare foundation
+    -- Create the WAI application and apply middlewares
+    appPlain <- toWaiAppPlain foundation
+    return $ logWare $ defaultMiddlewaresNoLogging appPlain
+
+makeLogWare :: App -> IO Application
```
Comment (MaxGabriel) [inline Application.hs]: I think this is supposed to return `IO Middleware`. Note: Middleware isn't currently being imported into this file, and `wai` isn't in the `build-depends` in the cabal file.

**Comment by MaxGabriel** [comment]
Made comments, otherwise LGTM. It is kind of noisy like you said, but debugging tests is a huge pain right now so seems like a win.

image

**Comment by gregwebs** [comment]
Thanks for doing the compilation for me! I wasn't sure what the workflow was for testing yesod scaffolding anymore and I was hoping that Travis CI would do the compilation.

**Comment by MaxGabriel** [comment]
@gregwebs That's a good point, we should probably have travis testing the scaffolding

**Comment by gregwebs** [comment]
comments are addressed. Unfortunately I still have not compiled it. Anyone know how to fix travis complaining about libgmp.so?

**Comment by MaxGabriel** [comment]
Potentially installing the gmp-devel package solves that? https://github.com/commercialhaskell/stack/issues/267

I'll try that out

**Comment by MaxGabriel** [comment]
@gregwebs This is good to merge, right? #76 duplicates the code to get yesod-test-1.5 working and I wanted to rebase that branch against this one to use your code instead.

**Comment by gregwebs** [comment]
Thanks again for figuring out Travis

**Pull request merged**